auth/handlers/query.go

```diff
@@ -58,8 +59,13 @@ func MakeQueryHandler(config *Config, protected []string) func(http.ResponseWrit
 		log.Printf("Validate %s => %d\n", resource, status)
 
 		if status == http.StatusTemporaryRedirect {
-			redirect := buildGitHubURL(config, "", config.Scope)
-			log.Printf("Go to %s\n", redirect.String())
+			var redirect *url.URL
+
+			if config.OAuthProvider == "gitlab" {
+				redirect = buildGitLabURL(config)
+			} else if config.OAuthProvider == "github" {
+				redirect = buildGitHubURL(config, "", config.Scope)
+			}
```

Comment:
Another place where a final `else` statement would help in the case that the given OAuthProvider is not one of the ones coded for

Reply:
I have implemented the same if statement as in `oauth2.go` in lines 64-73. Do you think that it's ok now?